packages/flutter/lib/src/painting/text_span.dart

```diff
@@ -300,31 +336,47 @@ class TextSpan extends DiagnosticableTree {
     }());
     return true;
   }
+  @override
+  bool debugAssertIsValidVisitor() {
+    if (children != null) {
+      for (InlineSpan child in children) {
+        if (child == null)
+          return false;
+      }
+    }
+    return true;
+  }
 
   /// Describe the difference between this text span and another, in terms of
   /// how much damage it will make to the rendering. The comparison is deep.
   ///
+  /// Comparing a [TextSpan] with a [WidgetSpan] will result in [RenderComparison.layout].
```

Comment:
Maybe for more clarity: "Comparing [InlineSpan] objects of different types, for example, comparing a [TextSpan] to a [WidgetSpan], always results in [RenderComparison.layout]."